plugin/evm/vm.go

```diff
@@ -308,110 +324,115 @@ func (vm *VM) Initialize(
 		log.Warn("Chain enabled `AllowFeeRecipients`, but chain config has not specified any coinbase address. Defaulting to the blackhole address.")
 	}
 
-	// Handle offline pruning
-	ethConfig.OfflinePruning = vm.config.OfflinePruning
-	ethConfig.OfflinePruningBloomFilterSize = vm.config.OfflinePruningBloomFilterSize
-	ethConfig.OfflinePruningDataDirectory = vm.config.OfflinePruningDataDirectory
-	if len(ethConfig.OfflinePruningDataDirectory) != 0 {
-		if err := os.MkdirAll(ethConfig.OfflinePruningDataDirectory, perms.ReadWriteExecute); err != nil {
-			log.Error("failed to create offline pruning data directory", "error", err)
-			return err
-		}
-	}
+	vm.genesisHash = ethConfig.Genesis.ToBlock(nil).Hash()
 
 	vm.chainConfig = g.Config
 	vm.networkID = ethConfig.NetworkId
 
+	lastAcceptedHash, err := vm.readLastAccepted()
+	if err != nil {
+		return err
+	}
+	log.Info("reading accepted block db", "lastAcceptedHash", lastAcceptedHash)
+
+	if err := vm.initializeMetrics(); err != nil {
+		return err
+	}
+
+	vm.networkCodec, err = message.BuildCodec()
+	if err != nil {
+		return err
+	}
+
+	// initialize peer network
+	vm.Network = peer.NewNetwork(appSender, vm.networkCodec, ctx.NodeID, vm.config.MaxOutboundActiveRequests)
+	vm.client = peer.NewClient(vm.Network)
+
+	if err := vm.initializeChain(lastAcceptedHash, ethConfig); err != nil {
```

Comment:
nit: could put `ethConfig` on `vm` like Coreth.

Reply:
yeah I thought it but it's not being used anywhere. so just let those being garbage collected.